src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc

```diff
@@ -0,0 +1,292 @@
+  auto cid = event.smart_object_correlation_id();
+
+  RaiseEvent4Response(response, cid);
+
+  unsubscribe_from_event(event.id());
+
+  const auto& it_queue_freezed = freezed_resumptions_.find(module_uid);
+  if (it_queue_freezed != freezed_resumptions_.end()) {
+    LOG4CXX_DEBUG(logger_, "Freezed resumptions found");
+    auto& queue_freezed = it_queue_freezed->second;
+    while (!queue_freezed.empty()) {
+      const auto& resumption_request = queue_freezed.front();
+      cid = resumption_request
+                .message[app_mngr::strings::params]
+                        [app_mngr::strings::correlation_id]
+                .asInt();
+      queue_freezed.pop();
+      RaiseEvent4Response(response, cid);
+    }
+    freezed_resumptions_.erase(it_queue_freezed);
+  }
+}
+
+void RCPendingResumptionHandler::ProcessNextFreezedResumption(
+    const ModuleUid& module_uid) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  auto pop_front_freezed_resumptions = [this](const ModuleUid& module_uid) {
+    const auto& it_queue_freezed = freezed_resumptions_.find(module_uid);
+    if (it_queue_freezed == freezed_resumptions_.end()) {
+      return std::shared_ptr<QueueFreezedResumptions::value_type>(nullptr);
+    }
+    auto& queue_freezed = it_queue_freezed->second;
+    if (queue_freezed.empty()) {
+      freezed_resumptions_.erase(it_queue_freezed);
+      return std::shared_ptr<QueueFreezedResumptions::value_type>(nullptr);
+    }
+    auto freezed_resumption =
+        std::make_shared<QueueFreezedResumptions::value_type>(
+            queue_freezed.front());
+    queue_freezed.pop();
+    if (queue_freezed.empty()) {
+      freezed_resumptions_.erase(it_queue_freezed);
+    }
+    return freezed_resumption;
+  };
+
+  auto freezed_resumption = pop_front_freezed_resumptions(module_uid);
+  if (!freezed_resumption) {
+    LOG4CXX_DEBUG(logger_, "Not freezed resumptions found");
+    return;
+  }
+
+  auto& resumption_request = *freezed_resumption;
+  auto subscription_request =
+      std::make_shared<smart_objects::SmartObject>(resumption_request.message);
+  const auto fid = GetFunctionId(subscription_request);
+  const auto cid =
+      resumption_request
+          .message[app_mngr::strings::params][app_mngr::strings::correlation_id]
+          .asInt();
+  LOG4CXX_DEBUG(logger_,
+                "Subscribing for event with function id: "
+                    << fid << " correlation id: " << cid);
+  subscribe_on_event(fid, cid);
+  pending_requests_.emplace(cid, *subscription_request);
+  LOG4CXX_DEBUG(logger_,
+                "Sending request with fid: " << fid << " and cid: " << cid);
+  application_manager_.GetRPCService().ManageHMICommand(subscription_request);
+}
+
+smart_objects::SmartObjectSPtr
+RCPendingResumptionHandler::CreateSubscriptionRequest(
+    const ModuleUid& module, const uint32_t correlation_id) const {
+  auto request =
+      RCHelpers::CreateUnsubscribeRequestToHMI(module, correlation_id);
```

Comment:
For creating subscribe requests we are creating unsubscrie request and change `kSubscribe` ? weired. Please add additional methof to RCHelpers